### PR TITLE
feat: Add a rake task for linting and sending results to an endpoint

### PR DESCRIPTION
**Pull request by prehnRA**
Add a rake task which can lint, then POST results to an API endpoint.

### CODE REVIEW

File: lib/tasks/pr_lint.rake [hunk 1]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+    lints = Linters.violations_for_pr(pr).group_by { |lint| lint.class }
+
+    payload = {
+      job_token: token,
+      line_results: (lints[Violation] || []).map(&:as_json),
+      pr_results: (lints[PrViolation] || []).map(&:message),
+    }
+
+    HTTParty.post(endpoint, {
+      body: payload.to_json,
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:37]: Layout/IndentFirstHashElement: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: lib/tasks/pr_lint.rake [hunk 2]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+      headers: {
+          'Content-Type' => 'application/json',
+          'Accept' => 'application/json',
+        },
+    })
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:42]: Layout/IndentFirstHashElement: Indent the right brace the same as the first position after the preceding left parenthesis.

File: lib/tasks/pr_lint.rake [hunk 3]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+    payload = {
+      job_token: token,
+      line_results: (lints[Violation] || []).map(&:as_json),
+      pr_results: (lints[PrViolation] || []).map(&:message),
+    }
+
+    HTTParty.post(endpoint, {
+      body: payload.to_json,
+      headers: {
+          'Content-Type' => 'application/json',
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:39]: Layout/IndentFirstHashElement: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.

File: lib/tasks/pr_lint.rake [hunk 4]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+          'Accept' => 'application/json',
+        },
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:41]: Layout/IndentFirstHashElement: Indent the right brace the same as the start of the line where the left brace is.

File: lib/tasks/pr_lint.rake [hunk 5]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
       puts "Please provide the URL of a GitHub PR to lint"
     end
   end
+
+  task :lint_to_endpoint, [] => :environment do |task, args|
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:20]: Lint/UnusedBlockArgument: Unused block argument - `task`. You can omit all the arguments if you don't care about them.

Comment (lintomat) [inline lib/tasks/pr_lint.rake:20]: Lint/UnusedBlockArgument: Unused block argument - `args`. You can omit all the arguments if you don't care about them.

File: lib/tasks/pr_lint.rake [hunk 6]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+    url = ENV['PR_URL'] || raise('Please provide the URL of the PR to lint')
+    endpoint = ENV['RESULTS_ENDPOINT'] || raise('Please provide a results endpoint where to send lints')
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:22]: Metrics/LineLength: Line is too long. [104/100]

File: lib/tasks/pr_lint.rake [hunk 7]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+    pr = PullRequest.from_url(url)
+    lints = Linters.violations_for_pr(pr).group_by { |lint| lint.class }
+
+    payload = {
+      job_token: token,
+      line_results: (lints[Violation] || []).map(&:as_json),
+      pr_results: (lints[PrViolation] || []).map(&:message),
+    }
+
+    HTTParty.post(endpoint, {
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:36]: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

File: lib/tasks/pr_lint.rake [hunk 8]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
       puts "Please provide the URL of a GitHub PR to lint"
     end
   end
+
+  task :lint_to_endpoint, [] => :environment do |task, args|
+    url = ENV['PR_URL'] || raise('Please provide the URL of the PR to lint')
+    endpoint = ENV['RESULTS_ENDPOINT'] || raise('Please provide a results endpoint where to send lints')
+    token = ENV['JOB_TOKEN'] || raise('Please provide the result endpoint token')
+
+    raise "Bad PR URL format" unless PullRequest::PR_URL_PATTERN =~ url
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:25]: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: lib/tasks/pr_lint.rake [hunk 9]
```diff
@@ -16,4 +16,29 @@ namespace :pr do
...
+
+    pr = PullRequest.from_url(url)
+    lints = Linters.violations_for_pr(pr).group_by { |lint| lint.class }
```
Comment (lintomat) [inline lib/tasks/pr_lint.rake:28]: Style/SymbolProc: Pass `&:class` as an argument to `group_by` instead of a block.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.io/relint/revelrylabs/lintron/354).

**Comment by prehnRA** [comment]
I'll fix up those lints momentarily.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.io/relint/revelrylabs/lintron/354).

**Comment by prehnRA** [comment]
Yep, this is setting the groundwork for being able to do that.

**Pull request merged**